couchbase/document_id.cxx

```diff
@@ -86,4 +89,18 @@ document_id::has_default_collection() const
 {
     return !use_collections_ || collection_path_ == "_default._default";
 }
+
+std::vector<std::byte>
+make_protocol_key(const document_id& id)
+{
+    std::vector<std::byte> key{};
+    if (id.is_collection_resolved()) {
```

Comment:
Probably a bit of an ancillary point, but the naming of this (is_collection_resolved) implies that this might incorrectly encode a document_id which is intended to have a collection_id, but which was not properly resolved yet?

Reply:
maybe it is bad naming, but it is there to cover the legacy case where we don't need to inject collection id